cohortextractor/pandas_utils.py

```diff
@@ -8,6 +8,15 @@ def dataframe_to_file(df, filename):
     elif filename.endswith(".feather"):
         df.to_feather(filename)
     elif filename.endswith(".dta") or filename.endswith(".dta.gz"):
+        # Make a shallow copy of the dataframe so we can replace certain
+        # columns without duplicating the entire thing in memory
+        df = df.copy(deep=False)
+        # Replace categorical columns with arrays of strings as Stata's
+        # categorical support isn't that great. See:
+        # https://github.com/opensafely-core/cohort-extractor/issues/534
+        for column, dtype in list(df.dtypes.items()):
+            if isinstance(dtype, pandas.Categorical):
+                df[column] = df[column].__array__()
```

Comment:
Apols I've not time to check myself but isn't `df[column].astype(str)` what we want here?

Reply:
Could well be. I don't really know Pandas. This seemed to do what I wanted though.

Reply:
Now worries, it's just the ergonomics of using dunder methods making me a bit sad, but /shrug